storage/store.go

```diff
@@ -1369,7 +1375,9 @@ func (s *store) DeleteImage(id string, commit bool) (layers []string, err error)
 		}
 		aContainerByImage := make(map[string]string)
 		for _, container := range containers {
-			aContainerByImage[container.ImageID] = container.ID
+			if container.ImageID != "" {
+				aContainerByImage[container.ImageID] = container.ID
```

Comment:
It should be fine if it's not set, but now that you mention it, this doesn't seem necessary to make this case work.  I'll drop it.